manager/perf.go

```diff
@@ -28,18 +28,59 @@ type PerfMapOptions struct {
 	// LostHandler - Callback function called when one or more events where dropped by the kernel
 	// because the perf ring buffer was full.
 	LostHandler func(CPU int, count uint64, perfMap *PerfMap, manager *Manager)
+
+	PerfMapStats PerfMapStats
+	// DumpHandler - Callback function called when manager.Dump() is called
+	// and dump the current state (human readable)
+	DumpHandler func(perfMap *PerfMap, manager *Manager) string
 }
 
 // PerfMap - Perf ring buffer reader wrapper
 type PerfMap struct {
-	manager     *Manager
-	perfReader  *perf.Reader
+	manager    *Manager
+	perfReader *perf.Reader
 
 	// Map - A PerfMap has the same features as a normal Map
 	Map
 	PerfMapOptions
 }
 
+// PerfMapStats contain perf map read/errors statistics
+// including per CPU map bytes and lost bytes
+type PerfMapStats struct {
+	ReadErrors  uint64
+	Closed      uint64
```

Comment:
What is the function of this field? I think it will only ever be `0` until the perfmap is stopped.

Reply:
Closed should be always 0, but could be useful to know in case of debugging if the map has been closed and the manager is still running. manager.PerfMaps would still contain the perfMap even if closed.

We should set manager/perf.go:157 m.state = paused (or introduce a new state closed would be more correct) before returning from the go routine.
What do you think ?